Realm/ObjectStore/object_store_exceptions.hpp

```diff
@@ -24,42 +24,78 @@
 #include <string>
 
 namespace realm {
+    class Property;
 
     class ObjectStoreException : public std::exception {
     public:
         enum class Kind {
```

Comment:
This was mainly to avoid a bunch of boilerplate code and make it easy to add new exceptions. I agree not using map would improve efficiency but I don't think performance is a concern in the cases we would be constructing any of these exceptions.